### DISCUSSION TITLE
Any way to force my installer to run only in Admin mode

### DISCUSSION THREAD

**Discussion by radu-solomon** [wixtoolset/issues#8874] [Questions]
We have an application that runs and installs our main app + tray icon. This all works. I have a CA that needs to set a registry key under HKLM\Software\OurApp as the installer starts. 

When i right click on the installer and run it, it all works and everything gets upgraded correctly.

If I run it just by double clicking on the MSI the CA fails with

> Error: System.UnauthorizedAccessException: Access to the registry key 'HKEY_LOCAL_MACHINE\SOFTWARE\OurApp' is denied.

which makes sense because at the stage it needs to run MSI did not ask for elevation yet so the CA is running as the user (i think i am understanding this right)

We are ok to stop the installer and show a message telling the user they must run as admin but everything i tried so far never stops the installer.

Is there a way to check this other than a CA with possibly a custom error message?

We are using WiX 5.0.2 and i have this in the wxs but it never triggers

`<Launch Condition="Privileged" Message="You need to be an administrator to install this product." />`

**Comment by robmen** [comment]
> set a registry key under HKLM\Software\OurApp as the installer starts.

Why?

**Reply by radu-solomon** [reply]
At this point, it's the only way to make the TrayIcon app stop before the upgrade without having the popup about files in use. 

I tried the CloseApplication, process kill, and multiple ways, and none of them currently works to stop the TrayIcon app before the RM flags it as holding files and poping-up the window about killing it, which then fails to kill it anyway and files are left in use resulting in the MSI exiting with reboot needed code. 

This Trayicon is controlled by a registry setting that tells it to run or not and it auto shuts down itself when this value is set to false. So the only workaround I found to make it work and release the DLLs it's using is this to date. If there is a better way happy to try it but I have been looking at all possible ideas and trying things for a while now with no luck.

**Reply by robmen** [reply]
The TrayIcon monitors registry keys but doesn't respond to WM_QUERY_ENDSESSION/WM_QUIT windows messages? That's pretty crazy.

**Reply by radu-solomon** [reply]
Agreed, but it is not something I can change in the short term as I do not have code access to it, and there is no time to rewrite it from scratch now. 

So, going back to my original question, is there a way to stop the installer and show an error message if it is not run as an admin? Regardless of why I need it in this particular case, this doesn't sound like an invalid use case where you need your installer to be run only in admin mode.
